### PR TITLE
[l10n] Add Hungarian (hu-HU) locale

### PR DESCRIPTION
**Pull request by noherczeg**
Please let me know if I missed anything!

### CODE REVIEW

**Comment by noherczeg** [comment]
Not sure what's up with CircleCI since I haven't changed any build scripts...

**Comment by noherczeg** [comment]
@LukasTy how can we fix the CircleCI issue? Can someone re-run the job for me maybe?

**Comment by LukasTy** [comment]
> @LukasTy how can we fix the CircleCI issue? Can someone re-run the job for me maybe?

We have recently seen a similar issue with CircleCI not being happy with some PRs or contributors for some reason. 🤷 
Adding a commit from a member usually fixes it.

Could you also follow [this discussion](https://github.com/mui/mui-x/pull/7645/files/be5776e61e37352e29c6c571b3b1c6714cad83a5#r1083345325) in order to finalize the changes for clock aria-label translations? 🤔

**Comment by noherczeg** [comment]
@LukasTy is it okay now?

File: packages/x-date-pickers/src/locales/huHU.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  dateRangePickerToolbarTitle: 'Dátumhatárok kiválasztása',
+
+  // Clock labels
+  clockLabelText: (view, time, adapter) =>
+    `${timeViews[view] ?? view} kiválasztása. ${
+      time === null
+        ? 'Nincs kiválasztva idő'
+        : `A kiválasztott idő ${adapter.format(time, 'fullTime')}`
+    }`,
+  hoursClockNumberText: (hours) => `${hours} ${timeViews.hours.toLowerCase()}`,
```
Comment (LukasTy) [inline packages/x-date-pickers/src/locales/huHU.ts:47]: This is an interesting approach to making sure the capitalization is correct, although, I'm not sure it's necessary to have it perfectly aligned for screen readers. 🙈

**Review by LukasTy — APPROVED** [review]
Thank you for your contribution! 💯 
The changes look great. 👌

**Pull request merged**